units/CMakeLists.txt

```diff
@@ -25,14 +25,20 @@ if(UNITS_HEADER_ONLY)
             units-header-only INTERFACE -DUNITS_NAMESPACE=${UNITS_NAMESPACE}
         )
     endif()
+    if(UNITS_BASE_TYPE)
+        target_compile_definitions(
+            units-header-only PUBLIC -DUNITS_BASE_TYPE=${UNITS_BASE_TYPE}
```

Comment:
Is there a reason for exposing the type used in the bitfield via this flag? How about, e.g.,
- `cmake -DUNITS_BASE_TYPE_BYTES=4` (or 8) or
- `cmake -DUNITS_WIDE_BASE_TYPE=On`?

Reply:
The main reason was consistency with some other libraries I interact with.    It isn't using the actual type you give it just the size of it, just the size but it does allow type operations on the define, whereas using the number would have to be translated.  Not that big of deal to do it differently but this way simplified a few implementation details slightly and was consistent with some other libraries.    So not a particularly strong reason.